aws/resource_aws_mwaa_environment.go

```diff
@@ -0,0 +1,753 @@
+	_, err := conn.DeleteEnvironment(&mwaa.DeleteEnvironmentInput{
+		Name: aws.String(d.Id()),
+	})
+	if err != nil {
+		if isAWSErr(err, mwaa.ErrCodeResourceNotFoundException, "") {
+			return nil
+		}
+
+		return err
+	}
+
+	return waitForMwaaEnvironmentDeletion(conn, d.Id())
+}
+
+func mwaaEnvironmentModuleLoggingConfigurationSchema(defaultEnabled bool) *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"cloud_watch_log_group_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  defaultEnabled,
+			},
+			"log_level": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  mwaa.LoggingLevelInfo,
+				ValidateFunc: validation.StringInSlice([]string{
+					mwaa.LoggingLevelCritical,
+					mwaa.LoggingLevelError,
+					mwaa.LoggingLevelWarning,
+					mwaa.LoggingLevelInfo,
+				}, false),
+			},
+		},
+	}
+}
+
+func expandMwaaEnvironmentLoggingConfiguration(l []interface{}) *mwaa.LoggingConfigurationInput {
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	input := &mwaa.LoggingConfigurationInput{}
+
+	m := l[0].(map[string]interface{})
+
+	if v, ok := m["dag_processing_logs"]; ok {
+		input.DagProcessingLogs = expandMwaaEnvironmentModuleLoggingConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := m["scheduler_logs"]; ok {
+		input.SchedulerLogs = expandMwaaEnvironmentModuleLoggingConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := m["task_logs"]; ok {
+		input.TaskLogs = expandMwaaEnvironmentModuleLoggingConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := m["webserver_logs"]; ok {
+		input.WebserverLogs = expandMwaaEnvironmentModuleLoggingConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := m["worker_logs"]; ok {
+		input.WorkerLogs = expandMwaaEnvironmentModuleLoggingConfiguration(v.([]interface{}))
+	}
+
+	return input
+}
+
+func expandMwaaEnvironmentModuleLoggingConfiguration(l []interface{}) *mwaa.ModuleLoggingConfigurationInput {
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	input := &mwaa.ModuleLoggingConfigurationInput{}
+	m := l[0].(map[string]interface{})
+
+	input.Enabled = aws.Bool(m["enabled"].(bool))
+	input.LogLevel = aws.String(m["log_level"].(string))
+
+	return input
+}
+
+func expandMwaaEnvironmentNetworkConfigurationCreate(l []interface{}) *mwaa.NetworkConfiguration {
+	m := l[0].(map[string]interface{})
+
+	return &mwaa.NetworkConfiguration{
+		SecurityGroupIds: expandStringSet(m["security_group_ids"].(*schema.Set)),
+		SubnetIds:        expandStringSet(m["subnet_ids"].(*schema.Set)),
+	}
+}
+
+func expandMwaaEnvironmentNetworkConfigurationUpdate(l []interface{}) *mwaa.UpdateNetworkConfigurationInput {
+	m := l[0].(map[string]interface{})
+
+	return &mwaa.UpdateNetworkConfigurationInput{
+		SecurityGroupIds: expandStringSet(m["security_group_ids"].(*schema.Set)),
+	}
+}
+
+func waitForMwaaEnvironmentCreation(conn *mwaa.MWAA, id string) error {
```

Comment:
Suggest splitting these waiters into separate packages, similar to [this comment](https://github.com/hashicorp/terraform-provider-aws/pull/17012#issuecomment-758871279).